### PR TITLE
get MSVC 2015+ healthy

### PR DESCRIPTION
**Pull request by hobu**
This gets the cmake-based builds with -DBUILD_SHARED_LIBS=ON healthy

https://trac.osgeo.org/geos/ticket/975

### CODE REVIEW

File: include/geos/geom/Location.h [hunk 1]
```diff
@@ -30,7 +30,8 @@ namespace geom { // geos::geom
  * For a description of the DE-9IM, see the
  * [OpenGIS Simple Features Specification for SQL](http://www.opengis.org/techno/specs.htm").
  */
-GEOS_DLL enum class Location : char {
+#undef Location
```
Comment (dbaston) [inline include/geos/geom/Location.h]: What is this `undef` needed for?

**Comment by dbaston** [comment]
Does it make sense to update the AzP file to build the shared library?

File: include/geos/geom/Location.h (hunk 1 above)
Comment (hobu) [reply include/geos/geom/Location.h]: killed debugging lint in 2387764a226199beb7d5b9bdc5c2cf81960b4f1c

**Comment by hobu** [comment]
I have no idea how and where stuff gets merged, but I think this is ready.

**Comment by dbaston** [comment]
Looks like @pramsey committed it as 1df959cc8a39807e63a09bff

**Pull request closed without merging**